myhoard/util.py

```diff
@@ -25,6 +25,29 @@
 ERR_TIMEOUT = 2013
 
 
+def get_tables_to_optimise(error_lines: list[str], *, log: Logger) -> Iterator[str]:
```

Comment:
I'd really prefer a non-lazy return type. Why?

I doubt there's a performance issue here and that generating an iterator helps us in that regard, and it's quite a common bug to try accidentally reusing an iterator.